sudo -u hdfs hadoop jar /usr/hdp/3.0.1.0-187/hadoop-mapreduce/hadoop-streaming-3.1.1.3.0.1.0-187.jar
 -mapper /user/admin/mapper1.py -reducer /user/admin/reducer1.py -input /user/admin/input/data.csv -output /user/admin/out
put/all_accidents  

sudo -u hdfs hadoop jar /usr/hdp/3.0.1.0-187/hadoop-mapreduce/hadoop-streaming-3.1.1.3.0.1.0-187.jar
 -mapper /user/admin/mapper2.py -reducer /user/admin/reducer2.py -input /user/admin/output/all_accidents -output /user/admin/output/make_year_count 

# COMMAND EXECUTION LOG:

sandbox-hdp login: root                                                                                  
root@sandbox-hdp.hortonworks.com's password:                                                             
Last login: Thu Nov 18 01:32:48 2021 from 172.18.0.2                                                     
[root@sandbox-hdp ~]# sudo -u hdfs hadoop jar /usr/hdp/3.0.1.0-187/hadoop-mapreduce/hadoop-streaming-3.1.
1.3.0.1.0-187.jar -mapper /user/admin/mapper1.py -reducer /user/admin/reducer1.py -input /user/admin/inpu
t/data.csv -output /user/admin/output/all_accidents
packageJobJar: [] [/usr/hdp/3.0.1.0-187/hadoop-mapreduce/hadoop-streaming-3.1.1.3.0.1.0-187.jar] /tmp/str
eamjob7267556136095828527.jar tmpDir=null                                                                
21/11/18 01:51:26 INFO client.RMProxy: Connecting to ResourceManager at sandbox-hdp.hortonworks.com/172.1
8.0.2:8050                                                                                               
21/11/18 01:51:27 INFO client.AHSProxy: Connecting to Application History server at sandbox-hdp.hortonwor
ks.com/172.18.0.2:10200                                                                                  
21/11/18 01:51:27 INFO client.RMProxy: Connecting to ResourceManager at sandbox-hdp.hortonworks.com/172.1
8.0.2:8050                                                                                               
21/11/18 01:51:27 INFO client.AHSProxy: Connecting to Application History server at sandbox-hdp.hortonwor
ks.com/172.18.0.2:10200                                                                                  
21/11/18 01:51:27 INFO mapreduce.JobResourceUploader: Disabling Erasure Coding for path: /user/hdfs/.stag
ing/job_1637173258135_0007                                                                               
21/11/18 01:51:28 INFO mapred.FileInputFormat: Total input files to process : 1                          
21/11/18 01:51:29 INFO mapreduce.JobSubmitter: number of splits:2                                        
21/11/18 01:51:29 INFO mapreduce.JobSubmitter: Submitting tokens for job: job_1637173258135_0007         
21/11/18 01:51:29 INFO mapreduce.JobSubmitter: Executing with tokens: []                                 
21/11/18 01:51:29 INFO conf.Configuration: found resource resource-types.xml at file:/etc/hadoop/3.0.1.0-
187/0/resource-types.xml                                                                                 
21/11/18 01:51:30 INFO impl.YarnClientImpl: Submitted application application_1637173258135_0007         
21/11/18 01:51:30 INFO mapreduce.Job: The url to track the job: http://sandbox-hdp.hortonworks.com:8088/p
roxy/application_1637173258135_0007/                                                                     
21/11/18 01:51:30 INFO mapreduce.Job: Running job: job_1637173258135_0007                                
21/11/18 01:51:45 INFO mapreduce.Job: Job job_1637173258135_0007 running in uber mode : false            
21/11/18 01:51:45 INFO mapreduce.Job:  map 0% reduce 0%                                                  
21/11/18 01:52:01 INFO mapreduce.Job: Task Id : attempt_1637173258135_0007_m_000001_0, Status : FAILED   
Error: java.lang.RuntimeException: Error in configuring object                                           
        at org.apache.hadoop.util.ReflectionUtils.setJobConf(ReflectionUtils.java:113)                   
        at org.apache.hadoop.util.ReflectionUtils.setConf(ReflectionUtils.java:79)                       
        at org.apache.hadoop.util.ReflectionUtils.newInstance(ReflectionUtils.java:137)                  
        at org.apache.hadoop.mapred.MapTask.runOldMapper(MapTask.java:462)                               
        at org.apache.hadoop.mapred.MapTask.run(MapTask.java:349)                                        
        at org.apache.hadoop.mapred.YarnChild$2.run(YarnChild.java:174)                                  
        at java.security.AccessController.doPrivileged(Native Method)                                    
        at javax.security.auth.Subject.doAs(Subject.java:422)                                            
        at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1730)          
        at org.apache.hadoop.mapred.YarnChild.main(YarnChild.java:168)                                   
Caused by: java.lang.reflect.InvocationTargetException                                                   
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)                                   
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)                 
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)         
        at java.lang.reflect.Method.invoke(Method.java:498)                                              
        at org.apache.hadoop.util.ReflectionUtils.setJobConf(ReflectionUtils.java:110)                   
        ... 9 more                                                                                       
Caused by: java.lang.RuntimeException: Error in configuring object                                       
        at org.apache.hadoop.util.ReflectionUtils.setJobConf(ReflectionUtils.java:113)                   
        at org.apache.hadoop.util.ReflectionUtils.setConf(ReflectionUtils.java:79)                       
        at org.apache.hadoop.util.ReflectionUtils.newInstance(ReflectionUtils.java:137)                  
        at org.apache.hadoop.mapred.MapRunner.configure(MapRunner.java:38)                               
        ... 14 more                                                                                      
Caused by: java.lang.reflect.InvocationTargetException                                                   
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)                                   
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)                 
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)         
        at java.lang.reflect.Method.invoke(Method.java:498)                                              
        at org.apache.hadoop.util.ReflectionUtils.setJobConf(ReflectionUtils.java:110)                   
        ... 17 more                                                                                      
Caused by: java.lang.RuntimeException: configuration exception                                           
        at org.apache.hadoop.streaming.PipeMapRed.configure(PipeMapRed.java:221)                         
        at org.apache.hadoop.streaming.PipeMapper.configure(PipeMapper.java:66)                          
        ... 22 more                                                                                      
Caused by: java.io.IOException: Cannot run program "/user/admin/mapper1.py": error=2, No such file or dir
ectory                                                                                                   
        at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)                                      
        at org.apache.hadoop.streaming.PipeMapRed.configure(PipeMapRed.java:208)                         
        ... 23 more                                                                                      
Caused by: java.io.IOException: error=2, No such file or directory                                       
        at java.lang.UNIXProcess.forkAndExec(Native Method)                                              
        at java.lang.UNIXProcess.<init>(UNIXProcess.java:247)                                            
        at java.lang.ProcessImpl.start(ProcessImpl.java:134)                                             
        at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)                                      
        ... 24 more                                                                                      
                                                                                                         
21/11/18 01:52:02 INFO mapreduce.Job: Task Id : attempt_1637173258135_0007_m_000000_0, Status : FAILED   
[2021-11-18 01:52:00.886]Exception from container-launch.                                                
Container id: container_e04_1637173258135_0007_01_000002                                                 
Exit code: 1                                                                                             
                                                                                                         
[2021-11-18 01:52:00.996]Container exited with a non-zero exit code 1. Error file: prelaunch.err.        
Last 4096 bytes of prelaunch.err :                                                                       
Last 4096 bytes of stderr :                                                                              
OpenJDK 64-Bit Server VM warning: INFO: os::commit_memory(0x00000000c1980000, 601358336, 0) failed; error
='Cannot allocate memory' (errno=12)                                                                     
                                                                                                         
                                                                                                         
[2021-11-18 01:52:01.007]Container exited with a non-zero exit code 1. Error file: prelaunch.err.        
Last 4096 bytes of prelaunch.err :                                                                       
Last 4096 bytes of stderr :                                                                              
OpenJDK 64-Bit Server VM warning: INFO: os::commit_memory(0x00000000c1980000, 601358336, 0) failed; error
='Cannot allocate memory' (errno=12)                                                                     
                                                                                                         
                                                                                                         
                                                                                                         
21/11/18 01:52:11 INFO mapreduce.Job: Task Id : attempt_1637173258135_0007_m_000001_1, Status : FAILED   
Error: java.lang.RuntimeException: Error in configuring object                                           
        at org.apache.hadoop.util.ReflectionUtils.setJobConf(ReflectionUtils.java:113)                   
        at org.apache.hadoop.util.ReflectionUtils.setConf(ReflectionUtils.java:79)                       
        at org.apache.hadoop.util.ReflectionUtils.newInstance(ReflectionUtils.java:137)                  
        at org.apache.hadoop.mapred.MapTask.runOldMapper(MapTask.java:462)                               
        at org.apache.hadoop.mapred.MapTask.run(MapTask.java:349)                                        
        at org.apache.hadoop.mapred.YarnChild$2.run(YarnChild.java:174)                                  
        at java.security.AccessController.doPrivileged(Native Method)                                    
        at javax.security.auth.Subject.doAs(Subject.java:422)                                            
        at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1730)          
        at org.apache.hadoop.mapred.YarnChild.main(YarnChild.java:168)                                   
Caused by: java.lang.reflect.InvocationTargetException                                                   
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)                                   
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)                 
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)         
        at java.lang.reflect.Method.invoke(Method.java:498)                                              
        at org.apache.hadoop.util.ReflectionUtils.setJobConf(ReflectionUtils.java:110)                   
        ... 9 more                                                                                       
Caused by: java.lang.RuntimeException: Error in configuring object                                       
        at org.apache.hadoop.util.ReflectionUtils.setJobConf(ReflectionUtils.java:113)                   
        at org.apache.hadoop.util.ReflectionUtils.setConf(ReflectionUtils.java:79)                       
        at org.apache.hadoop.util.ReflectionUtils.newInstance(ReflectionUtils.java:137)                  
        at org.apache.hadoop.mapred.MapRunner.configure(MapRunner.java:38)                               
        ... 14 more                                                                                      
Caused by: java.lang.reflect.InvocationTargetException                                                   
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)                                   
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)                 
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)         
        at java.lang.reflect.Method.invoke(Method.java:498)                                              
        at org.apache.hadoop.util.ReflectionUtils.setJobConf(ReflectionUtils.java:110)                   
        ... 17 more                                                                                      
Caused by: java.lang.RuntimeException: configuration exception                                           
        at org.apache.hadoop.streaming.PipeMapRed.configure(PipeMapRed.java:221)                         
        at org.apache.hadoop.streaming.PipeMapper.configure(PipeMapper.java:66)                          
        ... 22 more                                                                                      
Caused by: java.io.IOException: Cannot run program "/user/admin/mapper1.py": error=2, No such file or dir
ectory                                                                                                   
        at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)                                      
        at org.apache.hadoop.streaming.PipeMapRed.configure(PipeMapRed.java:208)                         
        ... 23 more                                                                                      
Caused by: java.io.IOException: error=2, No such file or directory                                       
        at java.lang.UNIXProcess.forkAndExec(Native Method)                                              
        at java.lang.UNIXProcess.<init>(UNIXProcess.java:247)                                            
        at java.lang.ProcessImpl.start(ProcessImpl.java:134)                                             
        at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)                                      
        ... 24 more                                                                                      
                                                                                                         
21/11/18 01:52:12 INFO mapreduce.Job: Task Id : attempt_1637173258135_0007_m_000000_1, Status : FAILED   
[2021-11-18 01:52:11.611]Exception from container-launch.                                                
Container id: container_e04_1637173258135_0007_01_000005                                                 
Exit code: 1                                                                                             
                                                                                                         
[2021-11-18 01:52:11.622]Container exited with a non-zero exit code 1. Error file: prelaunch.err.        
Last 4096 bytes of prelaunch.err :                                                                       
Last 4096 bytes of stderr :                                                                              
OpenJDK 64-Bit Server VM warning: INFO: os::commit_memory(0x00000000c1b80000, 601358336, 0) failed; error
='Cannot allocate memory' (errno=12)                                                                     
                                                                                                         
                                                                                                         
[2021-11-18 01:52:11.630]Container exited with a non-zero exit code 1. Error file: prelaunch.err.        
Last 4096 bytes of prelaunch.err :                                                                       
Last 4096 bytes of stderr :                                                                              
OpenJDK 64-Bit Server VM warning: INFO: os::commit_memory(0x00000000c1b80000, 601358336, 0) failed; error
='Cannot allocate memory' (errno=12)                                                                     
                                                                                                         
                                                                                                         
                                                                                                         
21/11/18 01:52:23 INFO mapreduce.Job: Task Id : attempt_1637173258135_0007_m_000001_2, Status : FAILED   
Error: java.lang.RuntimeException: Error in configuring object                                           
        at org.apache.hadoop.util.ReflectionUtils.setJobConf(ReflectionUtils.java:113)                   
        at org.apache.hadoop.util.ReflectionUtils.setConf(ReflectionUtils.java:79)                       
        at org.apache.hadoop.util.ReflectionUtils.newInstance(ReflectionUtils.java:137)                  
        at org.apache.hadoop.mapred.MapTask.runOldMapper(MapTask.java:462)                               
        at org.apache.hadoop.mapred.MapTask.run(MapTask.java:349)                                        
        at org.apache.hadoop.mapred.YarnChild$2.run(YarnChild.java:174)                                  
        at java.security.AccessController.doPrivileged(Native Method)                                    
        at javax.security.auth.Subject.doAs(Subject.java:422)                                            
        at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1730)          
        at org.apache.hadoop.mapred.YarnChild.main(YarnChild.java:168)                                   
Caused by: java.lang.reflect.InvocationTargetException                                                   
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)                                   
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)                 
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)         
        at java.lang.reflect.Method.invoke(Method.java:498)                                              
        at org.apache.hadoop.util.ReflectionUtils.setJobConf(ReflectionUtils.java:110)                   
        ... 9 more                                                                                       
Caused by: java.lang.RuntimeException: Error in configuring object                                       
        at org.apache.hadoop.util.ReflectionUtils.setJobConf(ReflectionUtils.java:113)                   
        at org.apache.hadoop.util.ReflectionUtils.setConf(ReflectionUtils.java:79)                       
        at org.apache.hadoop.util.ReflectionUtils.newInstance(ReflectionUtils.java:137)                  
        at org.apache.hadoop.mapred.MapRunner.configure(MapRunner.java:38)                               
        ... 14 more                                                                                      
Caused by: java.lang.reflect.InvocationTargetException                                                   
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)                                   
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)                 
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)         
        at java.lang.reflect.Method.invoke(Method.java:498)                                              
        at org.apache.hadoop.util.ReflectionUtils.setJobConf(ReflectionUtils.java:110)                   
        ... 17 more                                                                                      
Caused by: java.lang.RuntimeException: configuration exception                                           
        at org.apache.hadoop.streaming.PipeMapRed.configure(PipeMapRed.java:221)                         
        at org.apache.hadoop.streaming.PipeMapper.configure(PipeMapper.java:66)                          
        ... 22 more                                                                                      
Caused by: java.io.IOException: Cannot run program "/user/admin/mapper1.py": error=2, No such file or dir
ectory                                                                                                   
        at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)                                      
        at org.apache.hadoop.streaming.PipeMapRed.configure(PipeMapRed.java:208)                         
        ... 23 more                                                                                      
Caused by: java.io.IOException: error=2, No such file or directory                                       
        at java.lang.UNIXProcess.forkAndExec(Native Method)                                              
        at java.lang.UNIXProcess.<init>(UNIXProcess.java:247)                                            
        at java.lang.ProcessImpl.start(ProcessImpl.java:134)                                             
        at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)                                      
        ... 24 more                                                                                      
                                                                                                         
21/11/18 01:52:25 INFO mapreduce.Job: Task Id : attempt_1637173258135_0007_m_000000_2, Status : FAILED   
Error: java.lang.RuntimeException: Error in configuring object                                           
        at org.apache.hadoop.util.ReflectionUtils.setJobConf(ReflectionUtils.java:113)                   
        at org.apache.hadoop.util.ReflectionUtils.setConf(ReflectionUtils.java:79)                       
        at org.apache.hadoop.util.ReflectionUtils.newInstance(ReflectionUtils.java:137)                  
        at org.apache.hadoop.mapred.MapTask.runOldMapper(MapTask.java:462)                               
        at org.apache.hadoop.mapred.MapTask.run(MapTask.java:349)                                        
        at org.apache.hadoop.mapred.YarnChild$2.run(YarnChild.java:174)                                  
        at java.security.AccessController.doPrivileged(Native Method)                                    
        at javax.security.auth.Subject.doAs(Subject.java:422)                                            
        at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1730)          
        at org.apache.hadoop.mapred.YarnChild.main(YarnChild.java:168)                                   
Caused by: java.lang.reflect.InvocationTargetException                                                   
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)                                   
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)                 
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)         
        at java.lang.reflect.Method.invoke(Method.java:498)                                              
        at org.apache.hadoop.util.ReflectionUtils.setJobConf(ReflectionUtils.java:110)                   
        ... 9 more                                                                                       
Caused by: java.lang.RuntimeException: Error in configuring object                                       
        at org.apache.hadoop.util.ReflectionUtils.setJobConf(ReflectionUtils.java:113)                   
        at org.apache.hadoop.util.ReflectionUtils.setConf(ReflectionUtils.java:79)                       
        at org.apache.hadoop.util.ReflectionUtils.newInstance(ReflectionUtils.java:137)                  
        at org.apache.hadoop.mapred.MapRunner.configure(MapRunner.java:38)                               
        ... 14 more                                                                                      
Caused by: java.lang.reflect.InvocationTargetException                                                   
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)                                   
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)                 
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)         
        at java.lang.reflect.Method.invoke(Method.java:498)                                              
        at org.apache.hadoop.util.ReflectionUtils.setJobConf(ReflectionUtils.java:110)                   
        ... 17 more                                                                                      
Caused by: java.lang.RuntimeException: configuration exception                                           
        at org.apache.hadoop.streaming.PipeMapRed.configure(PipeMapRed.java:221)                         
        at org.apache.hadoop.streaming.PipeMapper.configure(PipeMapper.java:66)                          
        ... 22 more                                                                                      
Caused by: java.io.IOException: Cannot run program "/user/admin/mapper1.py": error=2, No such file or dir
ectory                                                                                                   
        at java.lang.ProcessBuilder.start(ProcessBuilder.java:1048)                                      
        at org.apache.hadoop.streaming.PipeMapRed.configure(PipeMapRed.java:208)                         
        ... 23 more                                                                                      
Caused by: java.io.IOException: error=2, No such file or directory                                       
        at java.lang.UNIXProcess.forkAndExec(Native Method)                                              
        at java.lang.UNIXProcess.<init>(UNIXProcess.java:247)                                            
        at java.lang.ProcessImpl.start(ProcessImpl.java:134)                                             
        at java.lang.ProcessBuilder.start(ProcessBuilder.java:1029)                                      
        ... 24 more                                                                                      
                                                                                                         
21/11/18 01:52:36 INFO mapreduce.Job:  map 100% reduce 100%                                              
21/11/18 01:52:37 INFO mapreduce.Job: Job job_1637173258135_0007 failed with state FAILED due to: Task fa
iled task_1637173258135_0007_m_000001                                                                    
Job failed as tasks failed. failedMaps:1 failedReduces:0 killedMaps:0 killedReduces: 0                   
                                                                                                         
21/11/18 01:52:37 INFO mapreduce.Job: Counters: 14                                                       
        Job Counters                                                                                     
                Failed map tasks=7                                                                       
                Killed map tasks=1                                                                       
                Killed reduce tasks=1                                                                    
                Launched map tasks=8                                                                     
                Other local map tasks=6                                                                  
                Data-local map tasks=2                                                                   
                Total time spent by all maps in occupied slots (ms)=345588                               
                Total time spent by all reduces in occupied slots (ms)=0                                 
                Total time spent by all map tasks (ms)=86397                                             
                Total vcore-milliseconds taken by all map tasks=86397                                    
                Total megabyte-milliseconds taken by all map tasks=88470528                              
        Map-Reduce Framework                                                                             
                CPU time spent (ms)=0                                                                    
                Physical memory (bytes) snapshot=0                                                       
                Virtual memory (bytes) snapshot=0                                                        
21/11/18 01:52:37 ERROR streaming.StreamJob: Job not successful!                                         
Streaming Command Failed!                                                                                